### PR TITLE
Looking at strange code

### PR DESCRIPTION
**Pull request by alexey-milovidov**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

File: src/Formats/JSONUtils.cpp [hunk 1]
```diff
@@ -72,7 +71,7 @@ namespace JSONUtils
             }
             else
             {
-                pos = find_first_symbols<opening_bracket, closing_bracket, '\\', '"'>(pos, in.buffer().end());
+                pos = find_first_symbols<opening_bracket, closing_bracket, '"'>(pos, in.buffer().end());
```
Comment (alexey-milovidov) [inline src/Formats/JSONUtils.cpp:74]: JSON cannot contain a backslash outside of a string literal.

File: src/Formats/JSONUtils.cpp [hunk 2]
```diff
@@ -89,19 +88,13 @@ namespace JSONUtils
...
-                        ++pos;
-                }
                 else if (*pos == '"')
                 {
                     quotes = true;
                     ++pos;
                 }
 
-                if (balance == 0)
+                if (!quotes && balance == 0)
```
Comment (alexey-milovidov) [inline src/Formats/JSONUtils.cpp:97]: If the string literal just started, it does not mean the row can end there.

**Comment by robot-ch-test-poll3** [comment]
*This is an automated comment for commit 6bb181ce55b10b1dcca265edea2d761e3e9e133b with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/58196/6bb181ce55b10b1dcca265edea2d761e3e9e133b/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/58196/6bb181ce55b10b1dcca265edea2d761e3e9e133b/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Comment by alexey-milovidov** [comment]
@Avogar, I'm not fully confident about these changes, but as all tests have passed, and they look logical anyway, I will merge. But I would appreciate double-checking from you.

**Pull request merged**